### PR TITLE
Create rao-commons module and move rao inputs methods to this module

### PR DESCRIPTION
**Pull request by TheMaskedTurtle**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Refactoring 

**What is the current behavior?** *(You can also link to an open issue here)*
No common blocks for the rao parts and inputs cleaning inside crac module

**What is the new behavior (if this is a feature change)?**
Create a new module for common methods dedicated to RAOs and start to move input cleaning inside this common part

### CODE REVIEW

File: ra-optimisation/rao-commons/pom.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    <artifactId>farao-rao-commons</artifactId>
+    <packaging>jar</packaging>
+    <name>Remedial Actions Optimisation Commons</name>
+    <description>Remedial actions optimisation common blocks to implement engines</description>
+    <dependencies>
+        <dependency>
+            <groupId>com.farao-community.farao</groupId>
+            <artifactId>farao-crac-api</artifactId>
+            <version>2.1.0-SNAPSHOT</version>
```
Comment (murgeyseb) [inline ra-optimisation/rao-commons/pom.xml:27]: ${project.version}

File: ra-optimisation/rao-commons/pom.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            <scope>compile</scope>
```
Comment (murgeyseb) [inline ra-optimisation/rao-commons/pom.xml:28]: Compile is default, you can remove scope

File: ra-optimisation/rao-commons/pom.xml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
Comment (murgeyseb) [inline ra-optimisation/rao-commons/pom.xml:32]: junit is a test dependency

File: ra-optimisation/pom.xml [hunk 4]
```diff
@@ -21,6 +21,7 @@
         <module>ra-optimisation-api</module>
         <module>rao-api</module>
         <module>search-tree-rao</module>
+        <module>rao-commons</module>
```
Comment (murgeyseb) [inline ra-optimisation/pom.xml:24]: alphabetical order

File: ra-optimisation/rao-commons/pom.xml (hunk 1 above)
Comment (TheMaskedTurtle) [reply ra-optimisation/rao-commons/pom.xml]: done

File: ra-optimisation/rao-commons/pom.xml (hunk 2 above)
Comment (TheMaskedTurtle) [reply ra-optimisation/rao-commons/pom.xml]: done

File: ra-optimisation/rao-commons/pom.xml (hunk 3 above)
Comment (TheMaskedTurtle) [reply ra-optimisation/rao-commons/pom.xml]: done

File: ra-optimisation/pom.xml (hunk 4 above)
Comment (TheMaskedTurtle) [reply ra-optimisation/pom.xml:23]: done

**Pull request merged**